plugins/Marketplace/Consumer.php

```diff
@@ -65,4 +72,18 @@ public function isValidConsumer()
 
         return $this->isValid;
     }
+
+    public function getConsumerLicenseStatusPluginWise(): array
+    {
+        if (!$this->consumerLicenseStatusPluginWise) {
+            $consumer = $this->getConsumer();
+            if (!empty($consumer['licenses'])) {
+                foreach ($consumer['licenses'] as $license) {
+                    $this->consumerLicenseStatusPluginWise[$license['plugin']['name']] = ['licenseStatus' => $license['status']];
```

Comment:
Do we need to have this as a nested array? If the variable holds just license statuses, we don't need it as an associative array, the status can be directly assigned to the plugin name key.

Reply:
@michalkleiner I added, if in future, if we ever want to get more info about license it would be easier, I can change it if you think it is totally unnecessary